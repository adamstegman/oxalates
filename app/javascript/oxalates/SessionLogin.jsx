import React from 'react';
import PropTypes from 'prop-types';

export const SessionLogin = ({ onStartAuthentication }) => {
  return (
    <button type="button"
            className="action login"
            onClick={onStartAuthentication}>
      Log in
    </button>
  );
};

SessionLogin.propTypes = {
  onStartAuthentication: PropTypes.func,
};
