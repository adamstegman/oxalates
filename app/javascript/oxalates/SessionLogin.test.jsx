import React from 'react';
import renderer from 'react-test-renderer';

import { SessionLogin } from './SessionLogin';

test('SessionLogin renders the login button', () => {
  const component = renderer.create(
    <SessionLogin onStartAuthentication={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
