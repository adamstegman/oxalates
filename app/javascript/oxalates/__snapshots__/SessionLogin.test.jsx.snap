// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionLogin renders the login button 1`] = `
<button
  className="action login"
  onClick={[Function]}
  type="button"
>
  Log in
</button>
`;
